internal/queue/consumer.go

```diff
@@ -0,0 +1,117 @@
+// Package queue implements the message queue functionality using RabbitMQ.
+package queue
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/Konstantsiy/image-converter/internal/config"
+	"github.com/Konstantsiy/image-converter/internal/converter"
+	"github.com/Konstantsiy/image-converter/internal/repository"
+	"github.com/Konstantsiy/image-converter/internal/storage"
+	"github.com/Konstantsiy/image-converter/pkg/logger"
+	"github.com/streadway/amqp"
+)
+
+// Consumer listens to the queue and processes outgoing messages.
+type Consumer struct {
+	client  *rabbitMQClient
+	repo    *repository.Repository
+	storage *storage.Storage
+}
+
+func NewConsumer(repo *repository.Repository, storage *storage.Storage, conf *config.Config) (*Consumer, error) {
+	client, err := initRabbitMQClient(conf)
+	if err != nil {
+		return nil, err
+	}
+	return &Consumer{repo: repo, storage: storage, client: client}, nil
+}
+
+// ListenToQueue listens to the queue channel in a separate goroutine.
+func (c *Consumer) ListenToQueue() error {
+	err := c.client.ch.Qos(1, 0, false)
+	if err != nil {
+		return fmt.Errorf("can't configure QoS: %w", err)
+	}
+
+	msgChannel, err := c.client.ch.Consume(c.client.queue.Name, "", false, false, false, false, nil)
+	if err != nil {
+		return fmt.Errorf("can't register channel: %w", err)
+	}
+
+	go func() {
+		for {
+			msg := <-msgChannel
+			go c.consumeFromQueue(&msg)
+		}
+	}()
```

Comment:
I would expect a `Listen` function to block execution, e.g. `http.ListenAndServe`. Ideally you want to give that choice to the caller so they can do this
```go
func main() {
    consumer.ListenToQueue()
{
```
or this
```go
func main() {
    go consumer.ListenToQueue()

    http.ListenAndServe()
}
```
It's a bit tricky since you need to return an `error` here, but doable.
Alternatively, if you know you don't have a use case where you need a listener to block, then you can launch a goroutine inside, but name it accordingly like `StartListener`/`LaunchListener`

Reply:
Now you have a backwards problem: your method is named: `LaunchListener()` but you don't __launch a listener__ inside, you __listen__. There's no `go listen()` or `go func() { for {...} }()`. There's only `for {...}`. So you need to pick:
1. Method is called `Listen` and it blocks and listens directly.
``` go
	for {
		msg := <-msgChannel
		go func() {
			err := c.consumeFromQueue(&msg)
		}()
	}
```
2. Method is called `LaunchListener` and it doesn't block and launches a listener, that works separately.
``` go
	go func() {
		for {
			msg := <-msgChannel
			go func() {
				err := c.consumeFromQueue(&msg)
			}()
		}
	}()
```